mits/configspec.ini

```diff
@@ -0,0 +1,78 @@
+
+# We offer up to this many complementary badges for each team member willing to
+# work the hour requirements (configured below).
+mits_badges_per_team = integer(default=3)
+
+# The number of hours we require in order for team members to qualify for their
+# complementary badges.  This is displayed but not yet enforced by the system.
+mits_hour_requirement = string(default="at least 4 hours per day")
+
+# When we tell people they can expect to hear back about whether their game has
+# been accepted.  This is a string rather than a specific date so we can be vague.
+mits_responses_expected = string(default="by the end of November")
+
+# Directory on the server where screenshots will be stored.
+mits_picture_dir = string(default="%(root)s/pictures")
+
+# Email address which will send MITS automated emails.
+mits_email = string(default="MAGFest Indie Tabletop Showcase <tabletop@magfest.org>")
+
+# Goes at the bottom of all MITS automated emails.
+mits_email_signature = string(default=" - Janus, MITS Department Head")
+
+
+[dates]
+
+# This is the date by which teams need to submit their applications.
+mits_submission_deadline = string(default="2017-10-30")
+
+# Teams can edit their applications even after submitting it until this deadline.
+mits_editing_deadline = string(default="2017-11-12")
+
+
+[enums]
+
+[[mits_access_level]]
+mits_admin = string(default="MITS Admin")
+
+[[mits_app_status]]
+pending = string(default="Pending")
+declined = string(default="Declined")
+accepted = string(default="Accepted")
+
+[[mits_phase]]
+development = string(default="Development")
+prototype = string(default="Playable Prototype")
+testing = string(default="Testing")
+near_completion = string(default="Near Completion")
+post_prod = string(default="Post-Production Advertising")
+
+[[mits_contact]]
+texting = string(default="Text message")
+calling = string(default="Phone call")
+
+[[mits_room_night]]
+thursday = string(default="Thursday night")
+friday = string(default="Friday night")
+saturday = string(default="Saturday Night")
+
+[[mits_schedule]]
+thursday_afternoon = string(default="Thursday noon - 4pm")
```

Comment:
This raises a big red flag. Does this really need to be so... manual? 😰

Reply:
I definitely gave that a lot of thought before deciding to go this route.  Given the UI this seemed to make the most sense in the end.

My thinking was that I really just wanted two columns of checkboxes for the applicants to check indicating when they were available to present.  The first is when they're available at all, and the second is when they would have enough people on hand to be able to make use of multiple tables.  (It's possible that there's a slicker UI than just a couple of columns of checkboxes for that kind of thing, like an interactive calendar or something, but columns of checkboxes jumped out at me as making the most sense.)

Given that, it seemed like the best implementation was just to use a ``MultiChoice`` column.

So at that point the question basically was, is it better to just manually type out all of the timeslots, or would it be cleaner to programmatically generate them.  The programmatic generation would look something like this, which I could put in ``config.py`` (and started to do so before I concluded that manually putting them here was better):

```
def textual_hour(dt):
    return ('midnight' if dt.hour == 0 else
            'noon' if dt.hour == 12 else dt.strftime('%-I%p').lower())

c.MITS_SCHEDULE_OPTS = []
start = c.EPOCH.replace(hour=16)
end = c.ESCHATON.replace(hour=16)
curr = start
while curr <= end:
    opt_name = curr.strftime('%A_%H').lower()
    opt_val = c.create_enum_val(opt_name)
    next = curr + timedelta(hours=4)
    desc = '{} {} - {}'.format(curr.strftime('%A'), textual_hour(curr), textual_hour(next))
    c.MITS_SCHEDULE_OPTS.append((opt_val, desc))
    curr = next
```

That's 15 somewhat dense lines of code (not counting the comments that I'd certainly want to write before including something like that in the codebase) to save 20 lines of easily understandable configuration, which is why I didn't go that route.  In theory the code above is more reusable, but in practice I suspect that the way we schedule teams is likely to change over time and so whatever we use this year will likely get thrown out anyway.

So that's why I did this icky-looking manual approach.  With all that being said, if either you or Rob feel like the code I wrote above is actually better than the manual config options, I can easily switch over to using that.  My intuition is that the current approach is cleaner, but I don't feel that strongly about it either way.